### PR TITLE
remove pulsar-client-schema

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Remove pulsar-client-schema module since its not really needed.  Merge the module with pulsar-client-original.  Reduce project complexity.

This change will also greatly reduce the size of some of the other packages e.g. datagenerator source NAR package from 35MB -> 7 MB
 
Please note I create a new module called pulsar-client-api to hold API interfaces/classes for pulsar.  This is needed for schema related interfaces/classes since many other modules depend on them and it doesn't make since to pull in pulsar-client-original just for the schema API.  After discussing with @merlimat, he agrees and recommends to move all API related interfaces/classes into a separate module, since it make it light weight and flexible for other to depend on as well as it works well with javadoc of which to current setup does not work because of shading.  In this PR, I only moved a subset of the schema API Interfaces into the pulsar-client-api module as a start.

This PR is read for review

### CODE REVIEW

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Worker.java [hunk 1]
```diff
@@ -22,22 +22,22 @@
 import java.util.List;
 import java.util.Map;
 
-import org.apache.pulsar.functions.proto.InstanceCommunication.Metrics;
+//import org.apache.pulsar.functions.proto.InstanceCommunication.Metrics;
```
Comment (merlimat) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Worker.java]: delete lines

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Worker.java [hunk 2]
```diff
@@ -22,22 +22,22 @@
...
+
     /**
      * Get all functions stats on a worker
      * @return
-     * @throws PulsarAdminException 
+     * @throws PulsarAdminException
      */
-    Metrics getFunctionsStats() throws PulsarAdminException;
-    
+//    Metrics getFunctionsStats() throws PulsarAdminException;
```
Comment (merlimat) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Worker.java]: delete

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/WorkerImpl.java [hunk 3]
```diff
@@ -35,7 +33,7 @@
 import org.apache.pulsar.client.admin.PulsarAdminException;
 import org.apache.pulsar.client.admin.Worker;
 import org.apache.pulsar.client.api.Authentication;
-import org.apache.pulsar.functions.proto.InstanceCommunication.Metrics;
+//import org.apache.pulsar.functions.proto.InstanceCommunication.Metrics;
```
Comment (merlimat) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/WorkerImpl.java]: delete

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/WorkerImpl.java [hunk 4]
```diff
@@ -52,21 +50,21 @@ public WorkerImpl(WebTarget web, Authentication auth) {
...
-           }
-           String jsonResponse = response.readEntity(String.class);
-           Metrics.Builder metricsBuilder = Metrics.newBuilder();
-           mergeJson(jsonResponse, metricsBuilder);
-           return metricsBuilder.build();
-       } catch (Exception e) {
-           throw getApiException(e);
-       }
-   }
+//    @Override
```
Comment (merlimat) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/WorkerImpl.java]: same

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/Field.java [hunk 5]
```diff
@@ -40,4 +33,42 @@
      */
     private final int index;
 
+    public Field(String name, int index) {
```
Comment (merlimat) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/Field.java]: Why not use Lombok? It will not be required at runtime anyway

File: pulsar-client-shaded/pom.xml [hunk 6]
```diff
@@ -58,11 +90,9 @@
               <minimizeJar>false</minimizeJar>
 
               <artifactSet>
-                <excludes>
-                  <exclude>org.apache.pulsar:pulsar-client-schema</exclude>
-                </excludes>
                 <includes>
                   <include>org.apache.pulsar:pulsar-client-original</include>
+                  <!--<include>org.apache.pulsar:avro-protobuf-shaded</include>-->
```
Comment (merlimat) [inline pulsar-client-shaded/pom.xml]: delete

File: pulsar-client-shaded/pom.xml [hunk 7]
```diff
@@ -72,7 +102,7 @@
                   <include>org.reactivestreams:reactive-streams</include>
                   <include>com.typesafe.netty:netty-reactive-streams</include>
                   <include>org.javassist:javassist</include>
-                  <include>com.google.protobuf:protobuf-java</include>
+                  <!--<include>com.google.protobuf:protobuf-java</include>-->
```
Comment (merlimat) [inline pulsar-client-shaded/pom.xml]: delete

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctionWorker.java [hunk 8]
```diff
@@ -57,22 +56,22 @@ void processArguments() throws Exception {
...
-        void runCmd() throws Exception {
-            String json = JsonFormat.printer().print(admin.worker().getFunctionsStats());
-            GsonBuilder gsonBuilder = new GsonBuilder();
-            if (indent) {
-                gsonBuilder.setPrettyPrinting();
-            }
-            System.out.println(gsonBuilder.create().toJson(new JsonParser().parse(json)));
-        }
-    }
+//    @Parameters(commandDescription = "dump all functions stats")
```
Comment (merlimat) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctionWorker.java]: is this related?

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/Field.java (hunk 5 above)
Comment (jerrypeng) [reply pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/Field.java]: Then I will have to pull in the lombok dependency.  Didn't feel appropriate.  Trying to keep pulsar-client-api as minimal as possible
